src/main/java/net/imglib2/converter/readwrite/WriteConvertedRandomAccessibleInterval.java

```diff
@@ -74,4 +75,31 @@ public WriteConvertedRandomAccess< A, B > randomAccess( final Interval interval
 	{
 		return new WriteConvertedRandomAccess< A, B >( sourceInterval.randomAccess( interval ), converterSupplier );
 	}
+
+	@Override
+	public B getType() {
+		return converterSupplier.get().convert( ( Sampler< ? extends A > ) new ConstantSampler( getSource().getType() ) );
+	}
+
+	private static class ConstantSampler< T > implements Sampler< T >
```

Comment:
Is duplicated, could be extracted and shared between `WriteConvertedRandomAccessibleInterval` and `WriteConvertedRandomAccessible`.